pizza_cutter_metadetect/masks.py

```diff
@@ -0,0 +1,450 @@
+        tile cross ra = 0 (crossra0).
+    gaia_stars : structured numpy array, optional
+        The GAIA star data indicating where the GAIA star mask holes are. The default
+        of None will result in no GAIA star mask holes.
+    healpix_nside : int, optional
+        The nisde of the healsparse mask. The default is 131072.
+
+    Returns
+    -------
+    hs_msk : healsparse map
+        The healsparse boolean map indicating what areas were masked and why. The
+        possible flag values are
+
+            MASK_NOSLICE = 2**0 - indicates no data for a given coadd slice
+            MASK_GAIA_STAR = 2**1 - indicates regions where a GAIA star hole exists
+    """
+    # We will build a coadd image of the mask bits and then convert them to
+    # healsparse at the end. We do things this way for a few reasons.
+    # 1. The coadd image of mask bits is the mask actually used by the code.
+    # 2. We do not have pure geometric primitives that are easy to write down
+    #    in healsparse due to various factors.
+    # 3. it makes the code a bit simpler since we don't have to combine more
+    #    than one representation of the mask to get the final one
+
+    # this image holds the mask bits
+    msk_img = np.zeros(coadd_dims, dtype=np.int32)
+
+    # first do GAIA star masks
+    # we don't union healsparse.Circle objects because we may need to
+    # apply a 90 degree rotation to the mask holes within each slice
+    # this is done in the GAIA masking functions we have and so we use those here
+    if gaia_stars is not None:
+        for slice_ind in tqdm.trange(len(obj_data), desc='making GAIA masks', ncols=79):
+            _mask_one_gaia_stars(
+                buffer_size=buffer_size,
+                central_size=central_size,
+                gaia_stars=gaia_stars,
+                symmetrize=preconfig["gaia_star_masks"]["symmetrize"],
+                coadd_dims=coadd_dims,
+                msk_img=msk_img,
+                scol=obj_data["orig_start_col"][slice_ind, 0],
+                srow=obj_data["orig_start_row"][slice_ind, 0],
+            )
+
+    # then do the slice masks for missing slices
+    # these are formally defined in pixels, not ra-dec, though we could likely
+    # use healsparse convex polygons instead of the pixels
+    for slice_ind in tqdm.tqdm(missing_slice_inds, desc='making slice masks', ncols=79):
+        _mask_one_slice(
+            buffer_size=buffer_size,
+            central_size=central_size,
+            coadd_dims=coadd_dims,
+            msk_img=msk_img,
+            scol=obj_data["orig_start_col"][slice_ind, 0],
+            srow=obj_data["orig_start_row"][slice_ind, 0],
+        )
+
+    # now lets flatten to bad pixels so far
+    # this is done to make the next bit of code easier to write
+    msk = msk_img != 0
+    x, y = np.meshgrid(np.arange(coadd_dims[1]), np.arange(coadd_dims[0]))
+    x = x[msk].ravel()
+    y = y[msk].ravel()
+    vals = msk_img[msk].ravel()
```

Comment:
Depending on the size of the slice this meshgrid can take a lot of memory and be slow.  You can get the indices directly from the `msk` indices with `unravel_index`: https://numpy.org/doc/stable/reference/generated/numpy.unravel_index.html

Reply:
good tip!

Reply:
I removed this call.